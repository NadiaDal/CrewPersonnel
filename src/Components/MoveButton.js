// @flow

import React from 'react';

type MoveButtonType = {
  onClickHandler: Function,
  onKeyPressHandler: Function,
  icon: string,
};

const MoveButton = ({ onClickHandler, onKeyPressHandler, icon }: MoveButtonType) => {
  return (
    <div onClick={onClickHandler} onKeyPress={onKeyPressHandler} role="button" tabIndex="0" style={styles.container}>
      {icon}
    </div>
  );
};

const styles = {
  container: {
    width: 30,
    padding: 10,
    backgroundColor: 'green',
  },
};

export default MoveButton;
