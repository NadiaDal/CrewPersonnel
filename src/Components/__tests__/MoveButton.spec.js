import React from 'react';
import renderer from 'react-test-renderer';
import MoveButton from '../MoveButton';

it('move to right renders correctly', () => {
  const identity = () => {};
  const tree = renderer
    .create(<MoveButton onClickHandler={identity} onKeyPressHandler={identity} icon={'>'} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('move to left renders correctly', () => {
  const identity = () => {};
  const tree = renderer
    .create(<MoveButton onClickHandler={identity} onKeyPressHandler={identity} icon={'<'} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
