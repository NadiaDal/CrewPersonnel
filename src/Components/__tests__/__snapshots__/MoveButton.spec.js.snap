// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`move to left renders correctly 1`] = `
<div
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "padding": "2px 5px",
      "width": 30,
    }
  }
  tabIndex="0"
>
  &lt;
</div>
`;

exports[`move to right renders correctly 1`] = `
<div
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "padding": "2px 5px",
      "width": 30,
    }
  }
  tabIndex="0"
>
  &gt;
</div>
`;
